cli/src/main.rs

```diff
@@ -40,7 +43,8 @@ struct Args {
     dev: bool,
 }
 
-static CTX: Lazy<Mutex<Context>> = Lazy::new(|| Mutex::new(Context::default()));
+static CTX: Lazy<Mutex<Context<PackageManager>>> =
+    Lazy::new(|| Mutex::new(Context::new_with_resolver(PackageManager {}).unwrap()));
```

Comment:
Didn't know you could do that! Thought `PackageManager` by itself only refers to the type (such as when using `Context<PackageManager>` and that you would have to instantiate it (even though it is a zst). But thanks!

Reply:
Never mind, it doesn't compile for me with your change:
```
❯ cargo build
   Compiling modmark v0.1.0 (/Users/cmdjojo/Documents/GitHub/modmark/cli)
error[E0423]: expected value, found struct `PackageManager`
  --> cli/src/main.rs:47:56
   |
47 |     Lazy::new(|| Mutex::new(Context::new_with_resolver(PackageManager).unwrap()));
   |                                                        ^^^^^^^^^^^^^^ help: use struct literal syntax instead: `PackageManager {}`
   |
  ::: cli/src/package.rs:20:1
   |
20 | pub struct PackageManager {}
   | ---------------------------- `PackageManager` defined here
```
Do you get it to compile?

Reply:
Oh, that's interesting! I thought there might be confusion between if you refer to the type or a value of the type, but apparently `let b: Box<Foo> = Box::new(Foo);` is fine (in your playground above). But apparently it works fine anyways! Do you think it is worth changing `pub struct PackageManager {}` for `pub struct PackageManager;`?

Reply:
I guess just a matter of taste so feel free to skip it!

Reply:
I think `pub struct PackageManger {}` is better since `pub struct PackageManger;` implies that it is a zero-sized type, and changing from a zero-sized-type to a non-zero-sized-type is a big "step" althrough it isn't as big of a step if we do `pub struct PackageManager {}`, then we are more "free" to add more fields if we see fit (which may become appropriate if it is to hold, let's say, a link to our registry)